projects/core/src/configurator/commons/facade/configurator-commons.service.ts

```diff
@@ -233,54 +239,72 @@ export class ConfiguratorCommonsService {
+          cartId: this.genericConfigUtilsService.getCartId(cartState.value),
+          productCode: productCode,
+          quantity: 1,
+          configId: configId,
+          ownerKey: ownerKey,
+        };
+        this.store.dispatch(
+          new ConfiguratorActions.AddToCart(addToCartParameters)
+        );
+      });
   }
 
   /**
    * Updates a cart entry, specified by the configuration.
    *
    * @param configuration - Configuration
    */
-  public updateCartEntry(configuration: Configurator.Configuration) {
-    this.activeCartService.requireLoadedCart().subscribe((cartState) => {
-      const parameters: Configurator.UpdateConfigurationForCartEntryParameters = {
-        userId: this.genericConfigUtilsService.getUserId(cartState.value),
-        cartId: this.genericConfigUtilsService.getCartId(cartState.value),
-        cartEntryNumber: configuration.owner.id,
-        configuration: configuration,
-      };
-
-      this.store.dispatch(new ConfiguratorActions.UpdateCartEntry(parameters));
-      this.checkForUpdateDone(configuration).subscribe((configAfterUpdate) => {
-        this.readConfigurationForCartEntry(configAfterUpdate.owner);
+  updateCartEntry(configuration: Configurator.Configuration): void {
+    this.activeCartService
+      .requireLoadedCart()
+      .pipe(take(1))
+      .subscribe((cartState) => {
+        const cartId = this.genericConfigUtilsService.getCartId(
+          cartState.value
+        );
+        const parameters: Configurator.UpdateConfigurationForCartEntryParameters = {
+          userId: this.genericConfigUtilsService.getUserId(cartState.value),
+          cartId: cartId,
+          cartEntryNumber: configuration.owner.id,
+          configuration: configuration,
+        };
+        this.store.dispatch(
+          new ConfiguratorActions.UpdateCartEntry(parameters)
+        );
       });
-    });
   }
 
-  checkForUpdateDone(
-    configuration: Configurator.Configuration
-  ): Observable<Configurator.Configuration> {
-    return this.store.pipe(
+  checkForActiveCartUpdateDone(): Observable<boolean> {
+    return this.activeCartService.requireLoadedCart().pipe(
+      take(1),
+      switchMap((cartState) =>
+        this.cartStore.pipe(
+          select(
+            MultiCartSelectors.getCartHasPendingProcessesSelectorFactory(
+              this.genericConfigUtilsService.getCartId(cartState.value)
+            )
+          ),
+          filter((hasPendingChanges) => !hasPendingChanges)
+        )
+      )
+    );
+  }
+
+  checkForUpdateDone(cartId: string): Observable<boolean> {
+    return this.cartStore.pipe(
       select(
-        ConfiguratorSelectors.getConfigurationFactory(configuration.owner.key)
-      ),
-      filter(
-        (configInUpdate) => configInUpdate.isCartEntryUpdatePending === false
+        MultiCartSelectors.getCartHasPendingProcessesSelectorFactory(cartId)
       ),
-      take(1)
+      filter((hasPendingChanges) => !hasPendingChanges)
     );
   }
 
```

Comment:
Is it possible to extract the Cart relevant methods to their own file?
configurator-cart.service.ts?

Reply:
Did it..